tests/combine_reports.py

```diff
@@ -0,0 +1,243 @@
+    }
+    for report_path, report_dict in test_reports.items():
+        protocol = report_dict["protocol"]
+        slave = report_dict["slave"]
+        if "timestamp" not in combined_report:
+            combined_report["timestamp"] = report_dict["timestamp"]
+            combined_report["hostname"] = report_dict["hostname"]
+        combined_report["time"] += report_dict["time"]
+        combined_report["tests"] = report_dict["tests"]
+
+        for classname in report_dict["testcases"].keys():
+            if classname not in combined_report["testcases"]:
+                combined_report["testcases"][classname] = {}
+            for name in report_dict["testcases"][classname].keys():
+                if name not in combined_report["testcases"][classname]:
+                    result = report_dict["testcases"][classname][name]["result"]
+                    if result != "PASSED":
+                        message = "PROTOCOL: {} - SLAVE: {} --> {} ({})".format(
+                            str(protocol),
+                            str(slave),
+                            report_dict["testcases"][classname][name]["result"],
+                            report_dict["testcases"][classname][name]["message"]
+                        )
+                        output =  "\n PROTOCOL: {} - SLAVE: {} --> {} \n {} \n {}".format(
+                            str(protocol),
+                            str(slave),
+                            report_dict["testcases"][classname][name]["result"],
+                            SEPARATOR,
+                            report_dict["testcases"][classname][name]["output"]
+                        )
+                    else:
+                        message = ""
+                        output = ""
+                    
+                    combined_report["testcases"][classname][name] = {
+                        "result": report_dict["testcases"][classname][name]["result"],
+                        "message": message,
+                        "output": output,
+                        "time": report_dict["testcases"][classname][name]["time"]
+                    }
+                else:
+                    if combined_report["testcases"][classname][name]["result"] == "ERROR":
+                        result = "ERROR" # ERROR before
+                    elif report_dict["testcases"][classname][name]["result"] == "ERROR":
+                        result = "ERROR" # ERROR now
+                    elif combined_report["testcases"][classname][name]["result"] == "FAILED":
+                        result = "FAILED" # FAILED before
+                    elif report_dict["testcases"][classname][name]["result"] == "FAILED":
+                        result = "FAILED" # FAILED now
+                    elif combined_report["testcases"][classname][name]["result"] == "SKIPPED":
+                        # SKIPPED before
+                        if combined_report["testcases"][classname][name]["result"] == "PASSED":
```

Comment:
I don't understand this if.
combined_report["testcases"][classname][name]["result"] can't be "SKIPPED" and "PASSED" at the same time